migrations/frontend/1674814035_add_unified_source_perms_table/up.sql

```diff
@@ -0,0 +1,15 @@
+CREATE TABLE IF NOT EXISTS src_permissions(
+    id SERIAL PRIMARY KEY,
+    user_id INT NULL REFERENCES users(id) ON DELETE CASCADE,
+    repo_id INT NOT NULL REFERENCES repo(id) ON DELETE CASCADE,
+    ext_account_id INT NULL REFERENCES user_external_accounts(id) ON DELETE CASCADE,
```

Comment:
```suggestion
    user_external_account_id INT NULL REFERENCES user_external_accounts(id) ON DELETE CASCADE,
```

Let's be explicit about references in the names. `<table_name in singular>_id` is the standard in our codebase, I think.